### PR TITLE
ocamlPackages.mirage-unix: 4.0.1 → 5.0.1; mirage-crypto: 0.10.5 → 0.10.7

### PR DESCRIPTION
**Pull request by vbgl**
###### Description of changes

Fixes & improvements:
https://github.com/mirage/mirage-unix/blob/v5.0.0/CHANGES.md#v500-2022-02-01
https://github.com/mirage/mirage-crypto/blob/v0.10.6/CHANGES.md#v0106-2022-03-29

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 191157` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>comby</li>
    <li>dune-release</li>
    <li>jackline</li>
    <li>ligo</li>
    <li>liquidsoap</li>
  </ul>
</details>

File: pkgs/development/ocaml-modules/mirage-unix/default.nix [hunk 1]
```diff
@@ -2,11 +2,11 @@
 
 buildDunePackage rec {
   pname = "mirage-unix";
-  version = "4.0.1";
+  version = "5.0.0";
```
Comment (superherointj) [inline pkgs/development/ocaml-modules/mirage-unix/default.nix:5]:
```suggestion
  version = "5.0.1";
```
`5.0.1` is available. Is `5.0.0` still desirable?

File: pkgs/development/ocaml-modules/mirage-crypto/default.nix [hunk 2]
```diff
@@ -1,23 +1,21 @@
...
 , ocaml-freestanding
 }:
 
 buildDunePackage rec {
-  minimumOCamlVersion = "4.08";
+  minimalOCamlVersion = "4.08";
 
   pname = "mirage-crypto";
-  version = "0.10.5";
+  version = "0.10.6";
```
Comment (superherointj) [inline pkgs/development/ocaml-modules/mirage-crypto/default.nix:10]:
```suggestion
  version = "0.10.7";
```
`0.10.7` is available. Is `0.10.6` still desirable?

File: pkgs/development/ocaml-modules/mirage-unix/default.nix (hunk 1 above)
Comment (superherointj) [reply pkgs/development/ocaml-modules/mirage-unix/default.nix:5]: > minimalOCamlVersion = "4.06";
Is this necessary?

Comment (vbgl) [reply pkgs/development/ocaml-modules/mirage-unix/default.nix:5]: What do you mean?

Comment (superherointj) [reply pkgs/development/ocaml-modules/mirage-unix/default.nix:5]: `minimalOCamlVersion` is missing but as it is `4.06` wondered if `4.06` is old enough to be disregarded. (Which is the case with `4.02`.) I have no opinion on this. You know better.

File: pkgs/development/ocaml-modules/mirage-crypto/default.nix (hunk 2 above)
Comment (vbgl) [reply pkgs/development/ocaml-modules/mirage-crypto/default.nix]: Added an update.

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 191157` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>comby</li>
    <li>dune-release</li>
    <li>jackline</li>
    <li>ligo</li>
    <li>liquidsoap</li>
  </ul>
</details>

**Review by superherointj — APPROVED** [review]

**Pull request merged**